### ISSUE TITLE
Allow switching between periods with and without text

### ISSUE THREAD

**Issue by miyukki** [google/shaka-player#715]
- What version of Shaka Player are you using?
  v2.0.6

  - Can you reproduce the issue with our latest release version?
    yes

  - Can you reproduce the issue with the latest code from `master`?
    yes

- Are you using the demo app or your own custom app?
  my custom app

  - If custom app, can you reproduce the issue using our demo app?
    not tried

- What browser and OS are you using?
  electron v1.6.1

- What did you do?

  - What content did you load?
    Load the MPD manifest written with some periods.
    One period has a video and audio. But next one has a video only.

- What did you expect to happen?
can play the content having different stream types between periods.

- What actually happened?
shaka raised INVALID_STREAMS_CHOSEN error when I played between period.

I found that shaka prevent different stream in StreamingEngine.
https://github.com/google/shaka-player/blob/fc46b29ed8b0781bf60165682f125470e2c04b22/lib/media/streaming_engine.js#L1687-L1708

Why shaka implemented such behaviour?

**Comment by TheModMaker** [comment]
We use MSE to play media.  We create a buffer for audio and video data and append data as we get it.  The browser pulls data from the buffers to play it in the `<video>` element.  If we switch from video+audio to just video, the browser will pause at the transition waiting for the content to buffer, but it doesn't know that the audio is never coming.

What we could do is reset MSE once we get to the transition, but then the transition won't be smooth since we need to delete everything and re-download it when the playhead gets there.  It would also be a lot more complicated to implement.  To make things simpler, we don't allow changing from video+audio to video-only.

**Comment by miyukki** [comment]
@TheModMaker Okay, I see.
Does shaka have the policy to prevent different stream types, or does it play if possible in the future?

**Comment by TheModMaker** [comment]
I don't understand your question, can you rephrase?  Are you asking if we support things other than audio/video/text, or are you asking about what the purpose of that code is?

**Comment by TheModMaker** [comment]
I should note that although we won't support changing video+audio to video-only, we should support changing between text and no text.  I was told that we don't support changing between Periods that do and do not have text streams.  That is something we should fix.

**Comment by miyukki** [comment]
@TheModMaker I understand. Thanks.

**Comment by joeyparrish** [comment]
Renamed and reopened to track switching between periods with and without text.

**Comment by joeyparrish** [comment]
This was just released in v2.0.7.
